src/scs/pki.md

```diff
@@ -0,0 +1,31 @@
+# Public Key Infrastructure
+
+In SCS context a public key infrastructure (PKI) term refers to a system for creation, storage, and distribution of digital certificates which are used to verify that a particular key belongs to a certain entity. PKI is creating and managing a hierarchical set of digital certificates which map public keys to certain entities. Storage and revocation methodologies are to be defined.
+
+The PK of SCS should consist of:
+  + A certificate authority (CA) responsible for storing, issuing and signing the digital certificates
+  + A registration authority (RA) responsible for requesting entity identity verification
+  + A central directory responsible for secure storage of the keys
+  + A Certificate Management System (CMS) managing access to stored certificates
+ 
+## Private Certificate Authority (PCA)
+
+PCA enables creation of private certificate authority hierarchies, consisting of root and subordinate CAs. It issues end-entity X.509 certificates, that are used for: 
+
+  + Encrypted TLS communication channels (data encryption in transit)
+  + Code and image signing
+
+PCA can be established in the cloud or on premise. Initially, openssl based on-prem solution is assumed, however, some of the target projects might consider using commercial cloud solutions.
```

Comment:
Yes, I need to get that out of my head somehow. Been using that for 10+ years, so it will be a hard journey, but I will learn one day. Thanks! :)

Reply:
But can we use "on-prem" or should it be only "on premises"?

Reply:
In CONTRIBUTING.md we have some kind of rule that sounds like:
_Avoid jargon and acronyms, if you can._
and 
_Do not assume that readers know everything you currently know._

In short, no :) And spelled with a hyphen *_*

Since these are not personal notes but official documentation that will be used as a basis for further development, we need to follow the rules. We do not know the readers' background. That's why it is better to use a way of writing that helps users understand a document the first time they read it.

So... I do not want to seem like a bore, but it is better not to use abbreviations and slang that are understandable to those who have been in the subject for a long time. If we assume that the text will be read by a person who is only studying security, it will be easier for him to google articles related to on-premises https://en.wikipedia.org/wiki/On-premises_software.